src/components/NeedsGameWorld.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#include <BsPrerequisites.h>
+#include <RTTI/RTTIUtil.hpp>
+#include <Scene/BsComponent.h>
+
+namespace REGoth
+{
+  class GameWorld;
+  using HGameWorld = bs::GameObjectHandle<GameWorld>;
+
+  /**
+   * Base-class for all components needing a GameWorld to work.
+   */
+  class NeedsGameWorld : public bs::Component
```

Comment:
Just stumbeled over the unusual class name. Maybe just personal taste. What about `GameWorldDependent`?

Reply:
It is unusual, but I feel it explains pretty nicely what you get if you use the component. There is the issue that a name like `GameWorldDependent` sounds like the component itself *is* a `GameWorld` which was extended with some other functionality. There could be a `GameWorldWithScriptInit` or `GameWorldIndoor` into whose naming scheme the component would incorrectly fall.

I agree classes should be named after a *thing*, but I'd have to research more about the naming of *components*, as they're not really a class in the traditional OOP way but more like traits assigned to an object and I would absolutely call a trait `NeedsGameWorld`.

Reply:
If you see them as trait, I can follow your arguments. I'm totally fine with the decision but was curious for the reason as I'm so used usual class naming conventions. The closest I would have guessed would be `NeedingGameWorld` or similar because `NeedsGameWorld` sounds more like the name of a (boolean) variable.

Reply:
We can follow "proper" OOP practices and call it `RequireGameWorldManagerInterfaceFactory`. :smile:  While working with Unity for university, my fellow students made sure to get a  `Manager` into each components name!

Jokes aside, I'll research some component naming when I get home!

Reply:
You surely meant the `RequireGameWorldManagerInterfaceFactoryVisitorObserver`. Because you want to make sure not to miss any design patterns and use them all in each circumstance 😀 I vote for this name

Reply:
Can't find proper guides on component naming. Only thing I have read was "Should be named after what it does", like `MeshCollider`. Now, what does that component do? I don't know anymore. If you're ok with it I'd like to keep the name. I kinda like it now :smile: